src/VirtualClient/VirtualClient.Actions/SysbenchOLTP/SysbenchOLTPClientExecutor.cs

```diff
@@ -0,0 +1,319 @@
+            await this.ExecuteCommandAsync<SysbenchOLTPClientExecutor>(sysbenchPath, this.sysbenchPrepareArguments, this.sysbenchDirectory, cancellationToken)
+                .ConfigureAwait(false);
+
+            await this.stateManager.SaveStateAsync<SysbenchOLTPState>($"{nameof(SysbenchOLTPState)}", state, cancellationToken);
+        }
+
+        private Task CaptureMetricsAsync(string results, DateTime startTime, DateTime endTime, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                results.ThrowIfNullOrWhiteSpace(nameof(results));
+
+                this.Logger.LogMessage($"{nameof(SysbenchOLTPClientExecutor)}.CaptureMetrics", telemetryContext.Clone()
+                    .AddContext("results", results));
+
+                try
+                {
+                    SysbenchOLTPMetricsParser parser = new SysbenchOLTPMetricsParser(results);
+                    this.Logger.LogMetrics(
+                        toolName: "Sysbench",
+                        scenarioName: "OLTP",
+                        startTime,
+                        endTime,
+                        parser.Parse(),
+                        metricCategorization: "Sysbench",
+                        scenarioArguments: this.sysbenchExecutionArguments,
+                        this.Tags,
+                        telemetryContext);
+                }
+                catch (Exception exc)
+                {
+                    throw new WorkloadException($"Failed to parse sysbench output: {results}.", exc, ErrorReason.InvalidResults);
+                }
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private Task ExecuteWorkloadAsync(IPAddress serverIpAddress, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            return this.Logger.LogMessageAsync($"{this.TypeName}.ExecuteWorkload", telemetryContext.Clone(), async () =>
+            {
+                string results = string.Empty;
+                this.StartTime = DateTime.UtcNow;
+                this.sysbenchExecutionArguments = $@"{this.Workload} --threads={this.Threads} --time={this.DurationSecs} --tables={this.NumTables} --table-size={this.RecordCount} --mysql-db={this.DatabaseName} --mysql-user=root --mysql-host={serverIpAddress}";
+
+                DateTime startTime = DateTime.UtcNow;
+                // gets executor arguments, combines with path & directory to get full command; metrics are stdout
+                string sysbenchPath = this.PlatformSpecifics.Combine(this.sysbenchDirectory, SysbenchOLTPClientExecutor.SysbenchFileName);
+                results += await this.ExecuteCommandAsync<SysbenchOLTPClientExecutor>(sysbenchPath, this.sysbenchExecutionArguments + " run", this.sysbenchDirectory, cancellationToken)
```

Comment:
If you don't expect the results to be reused, let's do == instead of +=. Just in case of weird bug we might encounter in the future